app/upgrades/v14_2/upgrades.go

```diff
@@ -33,8 +36,30 @@ func CreateUpgradeHandler(
 			}
 		}
 
+		// Add EIP contained in Shanghai hard fork to the extra EIPs
+		// in the EVM parameters. This enables using the PUSH0 opcode and
+		// thus supports Solidity v0.8.20.
+		logger.Info("adding EIP 3855 to EVM parameters")
+		err := EnableEIPs(ctx, ek, 3855)
+		if err != nil {
+			logger.Error("error while enabling EIPs", "error", err)
+		}
+
 		// Leave modules are as-is to avoid running InitGenesis.
 		logger.Debug("running module migrations ...")
 		return mm.RunMigrations(ctx, configurator, vm)
 	}
 }
+
+// EnableEIPs enables the given EIPs in the EVM parameters.
+func EnableEIPs(ctx sdk.Context, ek *evmkeeper.Keeper, eips ...int64) error {
+	evmParams := ek.GetParams(ctx)
+
+	for _, eip := range eips {
+		if !slices.Contains(evmParams.ExtraEIPs, eip) {
+			evmParams.ExtraEIPs = append(evmParams.ExtraEIPs, eip)
+		}
+	}
+
+	return ek.SetParams(ctx, evmParams)
```

Comment:
please validate the EIP

https://github.com/evmos/evmos/blob/main/x/evm/types/params.go#L157C10-L157C18

Reply:
That happens during `SetParams` already ([here](https://github.com/evmos/evmos/blob/v14.1.0/x/evm/keeper/params.go#L22-L25))? 🤔 

also I added unit tests that precisely check the case that we will be adding on testnet/mainnet and they pass, so I think we're good here.

Reply:
There is also a fail case for the error message that occurs when validating an invalid EIP here:
https://github.com/evmos/evmos/blob/malte/add-push0-eip/app/upgrades/v14_2/upgrades_test.go#L48-L54

Reply:
ah true, didn't know we validated the params when setting them